protocol/messenger_communities.go

```diff
@@ -532,6 +535,89 @@ func (m *Messenger) updateCommunitiesActiveMembersPeriodically() {
 	}()
 }
 
+func (m *Messenger) HandleCommunityGrantUpdated(state *ReceivedMessageState, update *protobuf.CommunityGrantUpdated, statusMessage *v1protocol.StatusMessage) error {
+	community, err := m.communitiesManager.GetByID(update.CommunityId)
+	if err != nil {
+		return err
+	}
+
+	err = m.communitiesManager.HandleCommunityGrant(community, update.Grant, update.Clock)
+	if err == communities.ErrGrantOlder || err == communities.ErrGrantExpired {
+		// Don't log an error for these cases
+		return nil
+	}
+	return err
+}
+
+func (m *Messenger) publishGrantForCommunityMemeber(community *communities.Community, memberKey string) error {
+	memberPubKey, err := common.HexToPubkey(memberKey)
+	if err != nil {
+		return err
+	}
+
+	grant, err := community.BuildGrant(memberPubKey, "")
+	if err != nil {
+		return err
+	}
+
+	update := &protobuf.CommunityGrantUpdated{
+		Clock:       uint64(time.Now().Unix()),
+		CommunityId: community.ID(),
+		Grant:       grant,
+	}
+
+	payload, err := proto.Marshal(update)
+	if err != nil {
+		return err
+	}
+
+	rawMessage := &common.RawMessage{
+		Payload:             payload,
+		Sender:              community.PrivateKey(),
+		SkipEncryptionLayer: true,
+		MessageType:         protobuf.ApplicationMetadataMessage_COMMUNITY_GRANT_UPDATED,
+		PubsubTopic:         shard.DefaultNonProtectedPubsubTopic(),
+	}
+
+	_, err = m.sender.SendPrivate(context.Background(), memberPubKey, rawMessage)
+	return err
+}
+
+func (m *Messenger) updateGrantsForControlledCommunities() {
+	// Resend grants for controlled communities every hour
+	ticker := time.NewTicker(grantUpdateInterval)
+
+	go func() {
+		for {
+			select {
+			case <-ticker.C:
```

Comment:
I believe the first tick should be almost immediate, to accommodate scenarios where the control node is only turned on for short duration, i.e. < few hours. See example: https://github.com/status-im/status-go/blob/759e5e5c7b7697168305eab6d69383a6914fd657/protocol/messenger_curated_communities.go#L43